luaui/Widgets/cmd_bar_hotkeys.lua

```diff
@@ -33,37 +25,82 @@ local function reloadWidgetsBindings()
 		Spring.SendCommands(command)
 	end
 end
 
-local function reloadBindings()
-	Spring.SendCommands("unbindall")
+local legacyToTxt = {
+	['luaui/configs/bar_hotkeys.lua']          = 'luaui/configs/hotkeys/default_keys.txt',
+	['luaui/configs/bar_hotkeys_mnemonic.lua'] = 'luaui/configs/hotkeys/mnemonic_keys.txt',
+	['luaui/configs/bar_hotkeys_60.lua']       = 'luaui/configs/hotkeys/default_keys_60pct.txt',
+	['luaui/configs/bar_hotkeys_grid.lua']     = 'luaui/configs/hotkeys/grid_keys.txt',
+	['luaui/configs/bar_hotkeys_grid_60.lua']  = 'luaui/configs/hotkeys/grid_keys_60pct.txt',
+	['bar_hotkeys_custom.lua']                 = 'uikeys.txt',
+}
 
-	currentLayout = Spring.GetConfigString("KeyboardLayout", 'qwerty')
-	currentKeybindingsFile = Spring.GetConfigString("KeybindingFile", "luaui/configs/bar_hotkeys.lua")
+local function replaceLegacyPreset()
+	local keyFile = Spring.GetConfigString("KeybindingFile", "noop")
 
-	loadBindings()
+	if keyFile == "noop" then return false end
 
-	reloadWidgetsBindings()
+	local newFormat = legacyToTxt[keyFile]
+
+	if not newFormat then return false end
+
+	Spring.Echo(os.clock(), newFormat, type(newFormat))
+
+	if keyFile == 'bar_hotkeys_custom.lua' then
+		Spring.Echo("BAR Hotkeys: bar_hotkeys_custom.lua found. This format is deprecated, a " .. newFormat .. " file was written to your bar folder")
+
+		if VFS.FileExists(keyFile) then
+			Spring.SendCommands("unbindall")
+			loadBindingsLegacy(keyFile)
+		end
+
+		Spring.SendCommands("keysave " .. newFormat)
+	else
+		Spring.SendCommands("keyreload " .. newFormat)
+	end
+
+	Spring.SetConfigString("KeybindingFile", newFormat)
+
+	return true
 end
 
-local function copyDefaultCustom()
-	if VFS.FileExists("bar_hotkeys_custom.lua") then return end
+local function reloadBindings()
+	currentLayout = Spring.GetConfigString("KeyboardLayout", 'qwerty')
+
+	local hasLegacy = replaceLegacyPreset()
 
-	local fileToCopy = Spring.GetConfigString("KeybindingFile", "luaui/configs/bar_hotkeys.lua")
-	fileToCopy = (fileToCopy ~= "bar_hotkeys_custom.lua") and fileToCopy or "luaui/configs/bar_hotkeys.lua"
+	currentKeybindingsFile = Spring.GetConfigString("KeybindingFile", keyConfig.keybindingPresets["Default"])
 
-	copyFileString("luaui/configs/bar_hotkeys.lua", "bar_hotkeys_custom.lua")
+	if not hasLegacy then
+		if VFS.FileExists(currentKeybindingsFile) then
+			Spring.SendCommands("keyreload " .. currentKeybindingsFile)
+			Spring.Echo("BAR Hotkeys: Loaded " .. (keyConfig.presetKeybindings[currentKeybindingsFile] or currentKeybindingsFile))
+		else
+			Spring.Echo("BAR Hotkeys: Did not find keybindings file " .. currentKeybindingsFile ..". Loading defaults")
+			Spring.SendCommands("keyreload " .. keyConfig.keybindingPresets["Default"])
+		end
+	end
+
+	reloadWidgetsBindings()
 end
 
 function widget:Initialize()
-	copyDefaultCustom()
 	reloadBindings()
 
+	Spring.SendCommands("keysave uikeys_backup.txt") -- save a backup of current keys
```

Comment:
This is a good idea for safety, but it would end up bloating the users `data` folder extremely fast, considering it outputs a backup on init. Maybe a system of rolling backups could be done, where only the most recent 5 will be saved or something.